node/src/mempool.rs

```diff
@@ -81,22 +85,63 @@ impl<N: Network, DB: database::DB, VM: vm::VMExecution>
         )
         .await?;
 
+        // Request mempool update from N alive peers
+        self.request_mempool(&network).await;
+
+        let idle_interval =
+            self.conf.idle_interval.unwrap_or(DEFAULT_IDLE_INTERVAL);
+
+        let mempool_expiry =
+            self.conf.mempool_expiry.unwrap_or(DEFAULT_EXPIRY_TIME);
+
+        // Mempool service loop
+        let mut on_idle_event = tokio::time::interval(idle_interval);
         loop {
-            if let Ok(msg) = self.inbound.recv().await {
-                match &msg.payload {
-                    Payload::Transaction(tx) => {
-                        let accept = self.accept_tx::<DB, VM>(&db, &vm, tx);
-                        if let Err(e) = accept.await {
-                            error!("{}", e);
-                            continue;
+            tokio::select! {
+                biased;
+                _ = on_idle_event.tick() => {
+                    info!(event = "mempool_idle", interval = ?idle_interval);
+
+                    let dur = time::SystemTime::now()
+                        .duration_since(UNIX_EPOCH)
+                        .expect("valid timestamp")
+                        .checked_sub(mempool_expiry)
+                        .expect("valid duration");
+
+                    // Remove expired transactions from the mempool
+                    db.read().await.update(|db| {
+                        let expired_txs = db.get_expired_txs(dur.as_secs())?;
```

Comment:
it's weird to see the expiration time passed as parameter
Also, `dur` is not really a clear variable name.

Reply:
Any suggestions?

Reply:
something like `expiration`/`expiration_time` would work.

or `validity`/`validity_time` if we want to see it as the timeframe where txs are still valid